### PR TITLE
WEBAPP-167: Test interaction of withFetcher and wrapped component

### PR DESCRIPTION
**Pull request by klinzo**

### CODE REVIEW

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ describe('CategoriesPage', () => {
 
   const language = 'en'
 
-  test('should render a Page if page has no children', () => {
+  test('should match snapshot and render a Page if page has no children', () => {
     const mockSetLanguageChangeUrls = jest.fn()
     const mockReplaceUrl = jest.fn()
```
Comment (maxammann) [inline src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: You don't need to mock functions if you don't use the mock. Just pass a () => {}

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js [hunk 2]
```diff
@@ -109,15 +109,15 @@ describe('CategoriesPage', () => {
...
                       replaceUrl={mockReplaceUrl} />
     )
 
     expect(wrapper).toMatchSnapshot()
   })
 
-  test('should render CategoryTiles if the path is the root category', () => {
+  test('should match snapshot and render CategoryTiles if the path is the root category', () => {
     const mockSetLanguageChangeUrls = jest.fn()
     const mockReplaceUrl = jest.fn()
```
Comment (maxammann) [inline src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: Same as above

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js [hunk 3]
```diff
@@ -292,89 +291,76 @@ describe('CategoriesPage', () => {
...
-      const mapLanguageToUrl = (language, id) => 'test' + language + id
-
-      const testUrls = {
-        en: 'testenundefined',
-        de: 'testde1235',
-        ar: 'testar1236'
+      const mapLanguageToUrl = (language, id) => `/${language}/${id}`
+      const languageChangeUrls = {
+        en: '/en/1235',
+        de: '/de/undefined',
```
Comment (maxammann) [inline src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: Should we maybe exclude this the de key if it's undefined?  Looks weird if we test for that invalid url.

File: src/routes/events/containers/__tests__/EventsPage.spec.js [hunk 4]
```diff
@@ -1,28 +1,22 @@
...
-import Payload from 'modules/endpoint/Payload'
-import EndpointBuilder from '../../../../modules/endpoint/EndpointBuilder'
-import EndpointProvider from '../../../../modules/endpoint/EndpointProvider'
-
-const mockSetLanguageChangeUrls = jest.fn()
 
 describe('EventsPage', () => {
   // we need UTC here, see https://medium.com/front-end-hacking/jest-snapshot-testing-with-dates-and-times-f3badb8f1d87
   // otherwise snapshot testing is not working
   moment.tz.setDefault('UTC') // fixme: leaks test
```
Comment (maxammann) [inline src/routes/events/containers/__tests__/EventsPage.spec.js]: Fix this here :)

File: src/routes/events/containers/__tests__/EventsPage.spec.js [hunk 5]
```diff
@@ -192,74 +192,59 @@ describe('EventsPage', () => {
...
-      const mapLanguageToUrl = (language, id) => 'test' + language + id
+      const mapLanguageToUrl = (language, id) => `/${language}/${id}`
 
-      const testUrls = {
-        en: 'testenundefined',
-        de: 'testde1235',
-        ar: 'testar1236'
+      const languageChangeUrls = {
+        en: '/en/1235',
+        de: '/de/undefined',
```
Comment (maxammann) [inline src/routes/events/containers/__tests__/EventsPage.spec.js]: Same as above.

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js (hunk 3 above)
Comment (klinzo) [reply src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: fixed

File: src/routes/events/containers/__tests__/EventsPage.spec.js (hunk 4 above)
Comment (klinzo) [reply src/routes/events/containers/__tests__/EventsPage.spec.js]: fixed

File: src/routes/events/containers/__tests__/EventsPage.spec.js (hunk 5 above)
Comment (klinzo) [reply src/routes/events/containers/__tests__/EventsPage.spec.js]: fixed

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js (hunk 2 above)
Comment (klinzo) [reply src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: done

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js (hunk 1 above)
Comment (klinzo) [reply src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: done

**Pull request merged**